### ISSUE TITLE
Terminal prepends typed characters to paste buffer

### ISSUE THREAD

**Issue by joechm** [sagemath/cloud#126]
Using firefox or chromium under ubuntu/linux: Select some text from terminal, call it "highlighted text". Next middle-mouse-button paste, everything looks good with "highlight text" on command line.

Run a command: $ ls

Middle click paste and get: $ lshighlighted text  

Note, "ls" is prepended to "highlighted text"

Occasionally bits of stdout get prepended, but I cannot seem to make it reproducible.

The prepended characters don't seem to be added to the X clipboard/primary selection; middle-click in another X11 application shows only "highlighted text" pasted.

**Comment by haraldschilly** [comment]
Yes, that behaviour is still there. The newly typed characters are prepended to the previously selected text in that middle-click clipboard buffer. At least, there is some odd side effects going on.
